### DISCUSSION TITLE
Discounts forces products to have a Category relation field

### DISCUSSION THREAD

**Discussion by jerome2710** [craftcms/commerce#2408] [Ideas] [answered]
**Description**
While this isn't exactly a bug, it does limit the functionality combined with our way of working. Products naturally don't have any fields and relations, besides the SKU/pricing stuff. To be able to group products, it is logical to add a relation field for categories of some sort.

We always tend to use sections for our categories, not categories themselves. Over the time, categories seem to have a few limiting factors for us, such as the lack of an expiry date (making them plannable) and not being able to be selected from a single relation field of multiple section entries (as they aren't entries, but categories).

While creating a new discount, it is possible to add a category condition. But this feature actually assumes you are using a category relation field in your products, which we are not. Therefore, we are unable to limit discounts on our categories, which are section entries.

I can imagine this scenario was not taken into consideration and I am unsure what could be a possible solution. Using categories instead of sections is a no-go for us, we've made that mistake more than once, stumbled into one or more of the limitations and ended up refactoring it back to sections after all. I'd like to open up a discussion in which we could share our thoughts and might come up with something better which would work for all of us.

**Additional info**
- Craft version: 3.4.27
- Commerce version: 3.1.10
- PHP version: 7.2.33

**Comment by lukeholder** [comment]
This will be solved with #80 when we give discounts support for a condition builder.

**Comment by jerome2710** [comment]
Thanks for mentioning that, I think I missed it. Looks great! Any idea where this stands on your roadmap?

**Comment by jerome2710** [comment]
Hi @lukeholder, just a quick mention, as another client just reached out to us with a similar question. As asked above, any idea when we could expect this? Thanks!

**Comment by lukeholder** [comment]
We will be allowing any entry or category element ID to be the relationship.

Adding this to sale conditions soon: https://github.com/craftcms/commerce/pull/2132

**Comment by lukeholder** [comment]
Since the discount condition is stored specifically as a category relation (I thought it was a element only relationship), we don't want to make this change in the short term since it would be a breaking change. Since Categories are going away (being migrated to entries) in Craft 4 we will wait for that to make the conversion.

**Comment by jerome2710** [comment]
Do you have any forecast when we might expect a public release of Craft 4 (not beta/RC) with this Commerce update, so we can keep our clients informed as well?

**Comment by lukeholder** [comment, marked as answer]
As of 4.3.0 discounts and sales support entry relationship conditions in prep for categories going away in Craft CMS in the future. https://github.com/craftcms/commerce/blob/develop/CHANGELOG.md#430---2023-09-13
